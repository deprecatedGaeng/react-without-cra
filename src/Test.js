import React, {useState} from "react";

const Test = ({name}) => {
  const [isClicked, setIsClicked] = useState(false);
  const handleClick = () => setIsClicked(true);
  
  return (
    <div>
      <button onClick={() => handleClick()}>{name}! CLICK ME!</button>
      <span>{isClicked && "CLICKED"}</span>
    </div>
  )
}

export default Test;