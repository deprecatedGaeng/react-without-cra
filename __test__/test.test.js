import React from 'react';
import Test from '../src/Test';
import { mount,configure,shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe("<Test />", () => {
  it("Snapshot Test", () => {
    const container = mount(<Test name="Cho" />);
    expect(container).toMatchSnapshot(); 
  })

  it('Props', () => {
    const container = mount(<Test name="Cho" />);
    expect(container.props().name).toBe('Cho');
  });

  it('Event Handler', () => {
    const container = mount(<Test name="Cho" />);
    const button = container.findWhere(node => node.type() === "button");
    button.simulate("click");

    const span = container.find('span');

    expect(span.text()).toBe('CLICKED');
  });
})