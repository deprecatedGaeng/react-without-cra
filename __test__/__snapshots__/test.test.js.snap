// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Test /> Snapshot Test 1`] = `
<Test
  name="Cho"
>
  <div>
    <button
      onClick={[Function]}
    >
      Cho
      ! CLICK ME!
    </button>
    <span />
  </div>
</Test>
`;
